File: 238-product-of-array-except-self/product-of-array-except-self.js
// 
var productExceptSelf = function(nums) {
   let leftMultiply = 1
   let rightMultiply = 1
   let left = []
   let right = []

   for (i = 0; i < nums.length; i++) {
        left[i] = leftMultiply
        leftMultiply *= nums[i]
   }

   for (let i = nums.length - 1; i >= 0; i--) {
    right[i] = rightMultiply * left[i]
    rightMultiply *= nums[i]
   } 

   return right
};

File: 238-product-of-array-except-self/product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        left = [None] * len(nums)
        right = [None] * len(nums)
        left_multiply = 1
        right_multiply = 1

        for i in range(len(nums)):
            left[i] = left_multiply
            left_multiply *= nums[i]

        for i in range(len(nums) - 1, -1, -1):
            right[i] = right_multiply * left[i]
            right_multiply *= nums[i]

        return right
        